### PR TITLE
dark plan strategy

### PR DESCRIPTION
**Pull request by chiahaoliu**
optimize fast shutter control sequence inside customized RunEngine

@CJ-Wright if you get time, can your review/merge it ? thanks.

### CODE REVIEW

File: xpdacq/tests/test_prun.py [hunk 1]
```diff
@@ -45,7 +45,7 @@ def setUp(self):
         # simulation objects
         glbl.area_det = SimulatedPE1C('pe1c', {'pe1_image': lambda: 5})
         glbl.temp_controller = be.motor
-        glbl.shutter = MagicMock()
+        glbl.shutter = be.Mover('motor', {'motor': lambda x: x}, {'x': 0})
```
Comment (CJ-Wright) [inline xpdacq/tests/test_prun.py]: Thank you!

Comment (chiahaoliu) [inline xpdacq/tests/test_prun.py]: I meant, he wrote it

File: xpdacq/xpdacq.py [hunk 2]
```diff
@@ -390,26 +395,31 @@ def __call__(self, sample, plan, subs=None, *,
...
-        if glbl.shutter_control:
-            plan = bp.pchain(bp.abs_set(sh, 1), plan, bp.abs_set(sh, 0))
+
         # Alter the plan to incorporate dark frames.
         if glbl.auto_dark:
             plan = dark_strategy(plan)
             plan = bp.msg_mutator(plan, _inject_qualified_dark_frame_uid)
+
+        # force to open shutter before scan and close it after
+        if glbl.shutter_control:
```
Comment (CJ-Wright) [inline xpdacq/xpdacq.py]: Is zero shutter closed?

Comment (CJ-Wright) [inline xpdacq/xpdacq.py]: I am worried that another number could mean an error or something else that is neither open nor closed.

Comment (CJ-Wright) [inline xpdacq/xpdacq.py]: Would it be more explicit to then say if not sixty? That way if the shutter is not fully open we open it the rest of the way.

**Pull request merged**